python-threatexchange/threatexchange/extensions/vpdq/vpdq_index.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Implementation of SignalTypeIndex abstraction for PDQ by wrapping
+hashing.pdq_faiss_matcher.
+"""
+
+import typing as t
+import vpdq
+from threatexchange.signal_type.index import (
+    PickledSignalTypeIndex,
+    VPDQIndexMatch,
+    IndexMatch,
+    T as IndexT,
+)
+from threatexchange.extensions.vpdq.vpdq_faiss import VPDQHashIndex
+from threatexchange.extensions.vpdq.vpdq_util import (
+    prepare_vpdq_feature,
+    VPDQ_QUALITY_THRESHOLD,
+    VPDQ_DISTANCE_THRESHOLD,
+)
+
+
+class VPDQIndex(PickledSignalTypeIndex[IndexT]):
+    """
+    Wrapper around the vpdq faiss index lib using VPDQHashIndex
+    """
+
+    def __init__(
+        self,
+        quality_threshold: int = VPDQ_QUALITY_THRESHOLD,
```

Comment:
👍

(ignorable): You may want to consider forcing this to be a kwarg by insert `*,` after self